benchmark/bandwagon_formula.py

```diff
@@ -0,0 +1,47 @@
+"""
+Bandwagon alpha formula.
+
+Idea:
+    We believe the 'wind' on the market.
+    The wind blows when the instrument rises/falls for consecutive x days,
+    and accumulated change rate is more than y.
+    We follow wind of instruments and positions are proportional to their
+    accumulated change rate (in the window).
+"""
+
+from thousandaire.alpha import BaseAlphaFormula
+from thousandaire.data_classes import Portfolio
+
+def catch_wind(data):
+    """
+    Do you feel the wind?
+    """
+    return (all([data[x].buy < data[x + 1].buy
+                 for x in range(-len(data), -1, 1)]) or
+            all([data[x].buy > data[x + 1].buy
+                 for x in range(-len(data), -1, 1)]))
+
+class BandWagonFormula(BaseAlphaFormula):
+    """
+    Formula for bandwagon.
+    """
+    def __init__(self, _startdate, data, parameters):
+        BaseAlphaFormula.__init__(self, _startdate, data, parameters)
+        self.rate = parameters['rate']
+        self.window = parameters['window']
+
+    def generate(self, _date, dataset):
+        """
+        Generate portfolio for the given date.
+        """
+        portfolio = Portfolio()
+        data = dataset['currency_price_tw']
+        for instrument in data:
+            used = data[instrument]
```

Comment:
The following comments are all based on my understanding that data is a dictionary. Please correct me if I am wrong :D
1. How about something like:
```for instrument_key, index_list in data.items():```? So that you don't have to iterate through "data" but call data[instrument] again.

2. Because data is a dictionary, "instrument" will be just a string(key). Personally I prefer "instrument_key" instead of "instrument" in order to be more specific on the naming for readability. At first sight, I thought "instrument" itself contains the index list. No harm to make it clear :D

3. Why is it named "used"? Looks like it is not used for storing any state.

Reply:
Named "used" just because I couldn't figure out a word to name it at that time. XD
It will be refactor.

Reply:
I think it's a good idea to write in the format ```for instrument_key, index_list in data.items():```
I think the ```instrument_key``` is something like a key to find which instruments we need (seems like the values of ```data[instrument_key]``` will be a list of string such as[ 'USD', 'JPY', 'EUR']) but the keys of data are truly the target trading instrument . Thus, I prefer ```instrument``` than ```instrument_key```.

Note:  input dataset is a dictionary when we type the name of target dataset (like 'currency_price_tw') as a key we will get the whole data of the dataset .
Each dataset has it's instruments which are our trading_targets.  ex.  Instruments of 'currency_price_tw' are 'USD', 'JPY', 'EUR', 'CNY' .... 
When I type  ```dataset['currency_price_tw']['USD']``` I will get the history data of  'USD' in 'currency_price_tw'
and I can use ```dataset['currency_price_tw']['USD'][-1]``` to get the yesterday's data.

Reply:
After rechecking alphas,  the names of some arguments are not consistent with their their type or meaning. 
That will make you confuse, I will refactor them.

Reply:
```seems like the values of data[instrument_key] will be a list of string such as[ 'USD', 'JPY', 'EUR']```
I mean if you use
```for instrument_key, index_list in data.items():```
you shouldnt have to do data[instrument_key]. For first loop you will get
 instrument_key = 'USD', index_list = // some history data
next loop:
 instrument_key = 'JPY', index_list = // some history data

Reply:
Oh! I knew this.  
I just mean that I prefer ```instrument``` than ``` instrument_key``` (the name of argument in for loop.)